### PR TITLE
chore: distribute commonjs version of assemblyscript

### PR DESCRIPTION
**Pull request by ajwootto**

### CODE REVIEW

**Comment by ajwootto** [comment]
Current dependencies on/for this PR:
* main
  * **PR #257** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/257" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/257?utm_source=stack-comment).

File: lib/shared/bucketing-assembly-script/index.cjs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const { instantiate } = require('./build/bucketing-lib.debug.commonjs')
+
+const url = new URL("build/bucketing-lib.release.commonjs.wasm", import.meta.url)
+
+exports.instantiate = async () => {
```
Comment (jonathannorris) [inline lib/shared/bucketing-assembly-script/index.cjs:5]: can you add some docs of what this is trying to do?

File: lib/shared/bucketing-assembly-script/index.cjs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    return await instantiate(
+        await (async () => {
+            try {
+                const source = await globalThis.fetch(url)
+                return await globalThis.WebAssembly.compileStreaming(source);
+            }
+            catch { return globalThis.WebAssembly.compile(await (await import("node:fs/promises")).readFile(url)); }
+        })(), {
```
Comment (jonathannorris) [inline lib/shared/bucketing-assembly-script/index.cjs:13]: why this empty object?

Comment (ajwootto) [reply lib/shared/bucketing-assembly-script/index.cjs:13]: just copied this from the ESM output the compiler was giving us 🤷

File: lib/shared/bucketing-assembly-script/index.cjs (hunk 1 above)
Comment (ajwootto) [reply lib/shared/bucketing-assembly-script/index.cjs:5]: I dont really know what its trying to do exactly, i just copied it from the ESM output. I can comment my best guess though

File: sdk/nodejs/__tests__/client.spec.ts [hunk 3]
```diff
@@ -1,6 +1,5 @@
 import { getBucketingLib } from '../src/bucketing'
 import { DVCClient } from '../src/client'
-jest.mock('../src/bucketing')
```
Comment (jonathannorris) [inline sdk/nodejs/__tests__/client.spec.ts:3]: I'm guessing we want to revert this?

Comment (ajwootto) [reply sdk/nodejs/__tests__/client.spec.ts:3]: it passes either way, I don't think it actually needs to be mocked here

File: scripts/npm-safe-publish.sh [hunk 4]
```diff
@@ -10,10 +10,10 @@ if [[ $# -eq 0 ]]; then
 fi
 
 PACKAGE=$1
-JQ_PATH=".version"
+JQ_PATH=".dependencies.\"${PACKAGE}\".version"
 
 NPM_SHOW="$(npm show "$PACKAGE" version)"
-NPM_LS="$(cat package.json | jq -r $JQ_PATH)"
+NPM_LS="$(npm ls "$PACKAGE" --json | jq -r $JQ_PATH)"
```
Comment (jonathannorris) [inline scripts/npm-safe-publish.sh:16]: is this the deploy script fix? why is it in here?

**Review by jonathannorris — APPROVED** [review]

**Review by jonathannorris — APPROVED** [review]

**Pull request merged**